tests/test_examples.py

```diff
@@ -0,0 +1,5 @@
+import subprocess
+
+
+def test_icmp_socket() -> None:
+    subprocess.run(["python3", "examples/icmp_socket.py"], check=True)
```

Comment:
Would it make sense to just import `ICMPSocket` from the test and call this like it is called in the tests main?

Reply:
I don't think so. Then we would have to keep the test in sync with the example. We should test a example in the same way as a user would use it.